pulsar-client-go/pulsar/c_producer.go

```diff
@@ -108,6 +108,14 @@ func createProducerAsync(client *client, options ProducerOptions, callback func(
 		C.pulsar_producer_configuration_set_compression_type(conf, C.pulsar_compression_type(options.CompressionType))
 	}
 
+	if options.SchemaInfo.Type != NONE {
+		C.pulsar_producer_configuration_set_schema_type(conf, C.pulsar_schema_type(options.SchemaInfo.Type),
+			C.CString(options.Name), C.CString(options.Schema))
```

Comment:
the `CString` instances will need to be released with `C.free()`

Reply:
Yes, otherwise it will cause a memory leak